_posts/2023-08-04-deploying-OIDC-webapp-on-openshift.adoc

```diff
@@ -0,0 +1,161 @@
+
+Alternatively you can link your pull secret so OpenShift can automatically use your pull secret to obtain the image when needed. You can learn more about handling secrets on OpenShift on https://docs.openshift.com/container-platform/4.10/openshift_images/managing_images/using-image-pull-secrets.html[their documentation]. 
+
+== Accessing the app
+
+Just like RH-SSO, you can find the url for your webapp by clicking on `Open URL` which will redirect you to the WildFly landing page. Now, let’s try accessing our application using <Simple-webapp-oidc-url>/simple-webapp-oidc. 
+
+Click on "Access Secured Servlet".
+
+Now, you’ll be redirected to RH-SSO's login page. If you click on the url on the search bar, you will see the scope values specified in the `redirect-uri` field with the different scope values separated by a `+`. You will also notice that a new scope value: `openid`. This indicates that we are going to be using OpenID Connect to authenticate the user. 
+
+Log in with `Alice` and the password that you set when configuring RH-SSO.
+
+Next, you’ll be redirected back to our application and you should see the "Secured Servlet" page. That means that we were able to successfully log in to our application using the RH-SSO OpenID provider!
+
+This page will display the Current Principal, and a list of claim values obtained using the scope values you configured. This is what it will look like: 
+```
+Secured Servlet
+Current Principal 'fc27fd69-3995-482c-9fad-0d45b838a804'
+
+Claims received using additional scope values:
+
+Using the "profile" scope, we got User's name: Alice Smith
+
+Using the "email" scope, we got email verified: true
+
+Using the "microprofile-jwt" scope, we got the user's groups: [default-roles-myrealm, offline_access, admin, uma_authorization, user]
+```
+
+Note that the value for Current Principal may be different, since that is the client secret. 
+
+Notice that there are no claims obtained using the `offline_access` scope. To learn more about what this scope value does, please refer to the https://openid.net/specs/openid-connect-core-1_0.html#OfflineAccess[ OpenID Documentation ]. 
+
+== Conclusion
+This post demonstrated how to deploy an OIDC secured WildFly application on OpenShift using Developer Sandbox. We also looked into how to configure additional scope values to request specific claim values. For more details on the elytron-oidc-client subsystem, please check out the https://docs.wildfly.org/29/Admin_Guide.html#Elytron_OIDC_Client[documentation] and for more details on WildFly on OpenShift, you can check out https://cloud.redhat.com/blog/getting-started-with-wildfly[this guide].
```

Comment:
Should mention here that this feature will be available in a future WildFly release.